Mathlib/NumberTheory/LSeries/HurwitzZetaOdd.lean

```diff
@@ -0,0 +1,592 @@
+  congrm (cexp ?_ * ?_) <;> ring
+
+lemma jacobiTheta₂''_neg_left (z τ : ℂ) : jacobiTheta₂'' (-z) τ = -jacobiTheta₂'' z τ := by
+  simp only [jacobiTheta₂'', jacobiTheta₂'_neg_left, jacobiTheta₂_neg_left,
+    neg_mul, neg_div, ← neg_add, mul_neg, neg_sq]
+
+lemma jacobiTheta₂'_functional_equation' (z τ : ℂ) :
+    jacobiTheta₂' z τ = (-2 * π) / (-I * τ) ^ (3 / 2 : ℂ) * jacobiTheta₂'' z (-1 / τ) := by
+  rcases eq_or_ne τ 0 with rfl | hτ
+  · rw [jacobiTheta₂'_undef _ (by simp), mul_zero, zero_cpow (by norm_num), div_zero, zero_mul]
+  have aux1 : (-2 * π : ℂ) / (2 * π * I) = I := by rw [div_eq_iff two_pi_I_ne_zero, mul_comm I,
+    mul_assoc _ I I, I_mul_I, neg_mul, mul_neg, mul_one]
+  rw [jacobiTheta₂'_functional_equation, ← mul_one_div _ τ, mul_assoc _ (cexp _), mul_comm (cexp _),
+    ← mul_assoc, (by rw [cpow_one, ← div_div, div_self (neg_ne_zero.mpr I_ne_zero)] :
+      1 / τ = -I / (-I * τ) ^ (1 : ℂ)), div_mul_div_comm,
+    ← cpow_add _ _ (mul_ne_zero (neg_ne_zero.mpr I_ne_zero) hτ), ← div_mul_eq_mul_div,
+    (by norm_num : (1 / 2  + 1 : ℂ) = 3 / 2), mul_assoc (1 / _), mul_assoc (1 / _),
+    ← mul_one_div (-2 * π : ℂ), mul_comm _ (1 / _), mul_assoc (1 / _)]
+  congr 1
+  rw [jacobiTheta₂'', div_add' _ _ _ two_pi_I_ne_zero, ← mul_div_assoc, ← mul_div_assoc,
+    ← div_mul_eq_mul_div (-2 * π : ℂ), mul_assoc, aux1, mul_div z (-1), mul_neg_one, neg_div τ z,
+    jacobiTheta₂_neg_left, jacobiTheta₂'_neg_left, neg_mul, ← mul_neg, ← mul_neg,
+    mul_div, mul_neg_one, neg_div, neg_mul, neg_mul, neg_div]
+  congr 2
+  rw [neg_sub, ← sub_eq_neg_add, mul_comm _ (_ * I), ← mul_assoc]
+
+/-- Odd Hurwitz zeta kernel (function whose Mellin transform will be the odd part of the completed
+Hurwitz zeta function). See `oddKernel_def` for the defining formula, and `hasSum_int_oddKernel`
+for an expression as a sum over `ℤ`.
+-/
+@[irreducible] def oddKernel (a : UnitAddCircle) (x : ℝ) : ℝ :=
+  (show Function.Periodic (fun a : ℝ ↦ re (jacobiTheta₂'' a (I * x))) 1 by
+    intro a; simp only [ofReal_add, ofReal_one, jacobiTheta₂''_add_left]).lift a
+
+lemma oddKernel_def (a x : ℝ) : ↑(oddKernel a x) = jacobiTheta₂'' a (I * x) := by
+  rw [oddKernel, Function.Periodic.lift_coe, ← conj_eq_iff_re, jacobiTheta₂''_conj, map_mul,
+    conj_I, neg_mul, neg_neg, conj_ofReal, conj_ofReal]
+
+lemma oddKernel_def' (a x : ℝ) : ↑(oddKernel ↑a x) = cexp (-π * a ^ 2 * x) *
+    (jacobiTheta₂' (a * I * x) (I * x) / (2 * π * I) + a * jacobiTheta₂ (a * I * x) (I * x)) := by
+  rw [oddKernel_def, jacobiTheta₂'', ← mul_assoc ↑a I x,
+    (by ring : ↑π * I * ↑a ^ 2 * (I * ↑x) = I ^ 2 * ↑π * ↑a ^ 2 * x), I_sq, neg_one_mul]
+
+lemma oddKernel_undef (a : UnitAddCircle) {x : ℝ} (hx : x ≤ 0) : oddKernel a x = 0 := by
+  induction' a using QuotientAddGroup.induction_on' with a'
+  rw [← ofReal_eq_zero, oddKernel_def', jacobiTheta₂_undef, jacobiTheta₂'_undef, zero_div, zero_add,
+    mul_zero, mul_zero] <;>
+  rwa [I_mul_im, ofReal_re]
```

Comment:
Since this occurs a number of times, it may make sense to add a
```lean
lemma Complex.im_I_mul_ofReal (x : ℝ) : (I * x).im = x := ...
```
(and the obvious variants) in, e.g., `Mathlib.Data.Complex.Basic` and then use it in this file.

Reply:
Good point, this already occurs 13 times in the `hurwitzZeta` code, I will add it